diff --git a/modules/contrib/search_autocomplete/src/Plugin/views/style/CallbackSerializer.php b/modules/contrib/search_autocomplete/src/Plugin/views/style/CallbackSerializer.php
index e088675..c3e36e4 100644
--- a/modules/contrib/search_autocomplete/src/Plugin/views/style/CallbackSerializer.php
+++ b/modules/contrib/search_autocomplete/src/Plugin/views/style/CallbackSerializer.php
@@ -141,7 +141,7 @@ public function renderGrouping($records, $groupings = array(), $group_rendered =
             // Extract group_id and transform it to machine name.
             $group_id = strtolower(Html::cleanCssIdentifier($this->getField($index, $group_field_name)));
             // Extract group displayed value.
-            $group_name = $this->getField($index, $group_field_name) . 's';
+            $group_name = $this->getField($index, $group_field_name);
           }
 
           // Create the group if it does not exist yet.
